File: android/app/src/main/java/jagdx/IDirect3DIndexBuffer.java
package jagdx;

import jaclib.peer.IUnknown;
import jaclib.peer.ti;

public final class IDirect3DIndexBuffer extends IUnknown {

    public IDirect3DIndexBuffer(ti arg0) {
        super(arg0);
    }

    @Override
    protected long a() {
        return super.a();
    }

    public boolean _Update(long arg0, int arg1, int arg2) { return false; }

    public int Lock(int arg0, int arg1, int arg2, GeometryBuffer arg3) { return 0; }

    public int Unlock() { return 0; }
}

File: src/jvmMain/kotlin/jagdx/IDirect3DIndexBuffer.kt
package jagdx

import jaclib.peer.IUnknown
import jaclib.peer.hb

actual class IDirect3DIndexBuffer actual constructor(arg0: hb?) : IUnknown(arg0) {
    actual override fun a(): Long {
        return super.a()
    }

    actual external fun Unlock(): Int

    actual external fun Lock(arg0: Int, arg1: Int, arg2: Int, arg3: GeometryBuffer?): Int

    /*private*/ actual external fun _Update(arg0: Long, arg1: Int, arg2: Int): Boolean
}
